tests/std/tests/P0896R4_views_take_while/test.cpp

```diff
@@ -0,0 +1,493 @@
+using move_only_view = test::range<Category, const int, test::Sized{is_random}, test::CanDifference{is_random},
+    IsCommon, test::CanCompare{derived_from<Category, forward_iterator_tag>},
+    test::ProxyRef{!derived_from<Category, contiguous_iterator_tag>}, test::CanView::yes, test::Copyability::move_only>;
+
+int main() {
+    // Validate views
+    { // ... copyable
+        constexpr span<const int> s{some_ints};
+        STATIC_ASSERT(test_one(s, only_less_than_three));
+        test_one(s, only_less_than_three);
+    }
+    { // ... move-only
+        test_one(move_only_view<input_iterator_tag, test::Common::no>{some_ints}, only_less_than_three);
+        test_one(move_only_view<forward_iterator_tag, test::Common::no>{some_ints}, only_less_than_three);
+        test_one(move_only_view<forward_iterator_tag, test::Common::yes>{some_ints}, only_less_than_three);
+        test_one(move_only_view<bidirectional_iterator_tag, test::Common::no>{some_ints}, only_less_than_three);
+        test_one(move_only_view<bidirectional_iterator_tag, test::Common::yes>{some_ints}, only_less_than_three);
+        test_one(move_only_view<random_access_iterator_tag, test::Common::no>{some_ints}, only_less_than_three);
+        test_one(move_only_view<random_access_iterator_tag, test::Common::yes>{some_ints}, only_less_than_three);
+    }
+
+    // Validate non-views
+    {
+        STATIC_ASSERT(test_one(some_ints, only_less_than_three));
+        test_one(some_ints, only_less_than_three);
+    }
+    {
+        vector vec(ranges::begin(some_ints), ranges::end(some_ints));
+        test_one(vec, only_less_than_three);
+    }
+    {
+        forward_list lst(ranges::begin(some_ints), ranges::end(some_ints));
+        test_one(lst, only_less_than_three);
+    }
+
+    // Validate a non-view borrowed range
+    {
+        constexpr span s{some_ints};
+        STATIC_ASSERT(test_one(s, only_less_than_three));
+        test_one(s, only_less_than_three);
+    }
+
+    // take_while/reverse interaction test
+    {
+        auto dwr_pipe = views::take_while(is_less_than_three) | views::reverse;
+        auto rdw_pipe = views::reverse | views::take_while(is_less_than_three);
+
+        auto r0  = some_ints | dwr_pipe;
+        using R0 = decltype(r0);
+        STATIC_ASSERT(ranges::bidirectional_range<R0> && ranges::view<R0>);
+        assert(ranges::equal(r0, views::reverse(only_less_than_three)));
+
+        auto r1  = some_ints | rdw_pipe;
+        using R1 = decltype(r1);
+        STATIC_ASSERT(ranges::bidirectional_range<R1> && ranges::view<R1>);
+        assert(ranges::equal(r1, views::reverse(only_less_than_three_reverse)));
+    }
+
+    { // empty range
+        constexpr span<const int, 0> empty{};
+        constexpr span<const int, 0> empty_expected{};
+        STATIC_ASSERT(test_one(empty, empty_expected));
+        test_one(empty, empty_expected);
+    }
+
+

```

Comment:
We must conserve empty lines whenever possible for our children's children:
```suggestion
```

Reply:
Finally someone thinks about the children :+1: